crates/broker/src/config.rs

```diff
@@ -126,6 +126,14 @@ pub struct MarketConf {
     ///
     /// Maximum number of concurrent lockin requests that can be processed at once
     pub max_concurrent_locks: Option<u32>,
+    /// Optional cache directory for storing downloaded images and inputs
+    /// if not set, files will be re-downloaded every time
+    pub cache_dir: Option<PathBuf>,
+    /// Optional S3 endpoint URL
+    pub s3_endpoint_url: Option<String>,
+    /// Optional IAM role for S3 access
+    /// The argument should take the form an Amazon Resource Name (ARN)
+    pub aws_assume_role_arn: Option<String>,
```

Comment:
Support for the IAM role is critical to take full advantage of the granular access control that AWS offers. This could also be an ENV variable (like all other AWS secrets), but since its name is not security relevant and having it in the config allows for better documentation, I went with the config, but would love to hear your thoughts on this.

Reply:
I didn't see this comment when I was doing my review, but in my opinion we should be consistent and stick with env variables. I think it is actually simpler to document, as with the current implementation we'll have to tell users "set your AWS_ACCESS_KEY_ID / SECRET in env, then set the role in config"

Reply:
Addressed in d22f06bf29da5784c3e1284e26b4c3f7eb10bf9b